Sources/Connection.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Connection.swift
+//  C7
+//
+//  Created by Scott Byrns on 4/1/16.
+//
+//
+
+public protocol Connection {
+    
+    associatedtype StatusType
+    
+    func open() throws
+    func close() -> Bool
+    
+    var connectionInfo: ConnectionInfo { get }
+    var status: StatusType { get }
+    var mostRecentError: ErrorProtocol? { get }
```

Comment:
In terms of mostRecentError, this was retained from the SQL projects version of this same protocol, I kept it so that the connection pool can check the errors on a given connection at any point and determine if the connection is healthy enough to use. Since a connection will be handed to the implementer from the pool and the context in which it is used is not the same as the one that provided it, the pool its self has little knowledge of what actions were taken and their effect. This property would allow for an inspection of the connection before handing a timed out or failed connection to another consumer.